ui/src/app/machines/components/ActionFormWrapper/TagForm/TagForm.js

```diff
@@ -58,7 +58,12 @@ export const TagForm = ({ setSelectedAction }) => {
       onSubmit={(values) => {
         if (values.tags && values.tags.length) {
           machinesToAction.forEach((machine) => {
-            dispatch(machineActions.tag(machine.system_id, values.tags));
+            dispatch(
+              machineActions.tag(
+                machine.system_id,
+                values.tags.map((tag) => tag.name)
+              )
+            );
```

Comment:
Just curious about the logic change here, was this broken previously?

Reply:
Ah, I changed the action to take an array of just the tag strings instead of the object of tags that was being supplied by the tag component. This made it more generic and the action no longer relied on the shape of the data from that one component.

Was:

https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/base/actions/machine/machine.js#L240

Became:

https://github.com/canonical-web-and-design/maas-ui/pull/1876/files/a5a93272a56fa859c77417c66943b1215be51118#diff-9115cdee452bde71414a086f6c8ea008af76fd08918e4b43c75d6caf955157ffR269